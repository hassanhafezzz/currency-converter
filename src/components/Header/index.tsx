import React, { FC } from 'react';
import classNames from 'classnames/bind';

import logo from '../../imgs/logo.svg';
import styles from './styles.module.css';

const cx = classNames.bind(styles);

const Header: FC = () => {
  return (
    <header className={cx('header')}>
      <nav>
        <h1 className={cx('brand')}>
          <a href="/">
            <span className={cx('visually-hidden')}>Currency converter</span>
            <img
              src={logo}
              className={cx('logo')}
              alt="currency converter logo"
            />
          </a>
        </h1>
      </nav>
    </header>
  );
};

export default Header;
