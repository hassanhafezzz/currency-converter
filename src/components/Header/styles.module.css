@value utils: '../../styles/utils.css';

.header {
  display: flex;
  align-items: center;
  padding: 0 20px;
  background-color: var(--color-white);
  height: 60px;
  box-shadow: 0 1px 10px 2px var(--color-gray-100);
}

.logo {
  height: 30px;
}

.visually-hidden {
  composes: visually-hidden from utils;
}
