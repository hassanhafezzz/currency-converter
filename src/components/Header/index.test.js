import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from './index';

test('header has app logo', () => {
  render(<Header />);
  const logo = screen.getByAltText(/logo/i);
  expect(logo).toBeInTheDocument();
});

test('add snapshot for header ', () => {
  const { container, asFragment } = render(<Header />);
  expect(asFragment(container)).toMatchSnapshot();
});
