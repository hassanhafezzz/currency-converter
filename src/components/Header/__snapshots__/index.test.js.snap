// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add snapshot for header  1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <nav>
      <h1
        class="brand"
      >
        <a
          href="/"
        >
          <span
            class="visually-hidden"
          >
            Converto
          </span>
          <img
            alt="Converto app logo"
            class="logo"
            src="logo.svg"
          />
        </a>
      </h1>
    </nav>
  </header>
</DocumentFragment>
`;
